runtime/jcl/common/shared.c

```diff
@@ -578,6 +579,11 @@ getCpeTypeForProtocol(char* protocol, jsize protocolLen, const char* pathChars,
 				}
 			}
 		}
+		/* Check again if the path is a file. There may be a jar file that does not use the expected .jar endig. */
+		if (EsIsFile == j9file_attr(pathChars)) {
```

Comment:
I guess we will still return incorrect type if a directory name ends with `.jar`.

Reply:
Good point. I modified the change so the directory type is only set if j9file_attr returns EsIsDir.

Reply:
I also added a test.